### ISSUE TITLE
Unable to build project on Xcode 15 with Firebase SPM package

### ISSUE THREAD

**Issue by tarrouye** [firebase/firebase-ios-sdk#11759]
### Description

When trying to build our project under any of the Xcode 15 betas, including the latest beta 8 at the time of writing, we are facing a build issue relating to the new framework signatures. 

`*/Xcode/DerivedData/*/SourcePackages/artifacts/grpc-binary/gRPC-C++/gRPC-C++.xcframework:1:1 The signature of “gRPC-C++.xcframework” cannot be verified.`

The closest relevant issue I could find is #11401 , which was closed as fixed, but we still face this issue. 

We use the following Firebase frameworks: 
<img width="167" alt="image" src="https://github.com/firebase/firebase-ios-sdk/assets/5017270/31b7fc97-3772-4179-858a-13d33ae48526">

### Firebase SDK Version

10.7, 10.8, 10.9, 10.10

### Xcode Version

15.0 beta 8

### Installation Method

Swift Package Manager

### Firebase Product(s)

Analytics, Crashlytics, DynamicLinks, Firestore, Messaging, Remote Config

### Targeted Platforms

iOS, macOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

{
  "pins" : [
    {
      "identity" : "abseil-cpp-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/abseil-cpp-binary.git",
      "state" : {
        "revision" : "bfc0b6f81adc06ce5121eb23f628473638d67c5c",
        "version" : "1.2022062300.0"
      }
    },
    {
      "identity" : "alamofire",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/Alamofire/Alamofire.git",
      "state" : {
        "revision" : "8dd85aee02e39dd280c75eef88ffdb86eed4b07b",
        "version" : "5.6.2"
      }
    },
    {
      "identity" : "appauth-ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/openid/AppAuth-iOS.git",
      "state" : {
        "revision" : "71cde449f13d453227e687458144bde372d30fc7",
        "version" : "1.6.2"
      }
    },
    {
      "identity" : "charts",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/danielgindi/Charts",
      "state" : {
        "branch" : "master",
        "revision" : "2c77b2b8a4ccb82a36260b44b2c794683cc6f17b"
      }
    },
    {
      "identity" : "cryptoswift",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/krzyzanowskim/CryptoSwift.git",
      "state" : {
        "revision" : "039f56c5d7960f277087a0be51f5eb04ed0ec073",
        "version" : "1.5.1"
      }
    },
    {
      "identity" : "facebook-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/facebook/facebook-ios-sdk.git",
      "state" : {
        "revision" : "c46ba480e9b176a950c428b1a1ec3fd57810f8fe",
        "version" : "11.2.1"
      }
    },
    {
      "identity" : "firebase-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/firebase-ios-sdk.git",
      "state" : {
        "revision" : "5034479ca0c4f32f299677fb0ba1fe4d3e3e20b4",
        "version" : "10.10.0"
      }
    },
    {
      "identity" : "floatingpanel",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/scenee/FloatingPanel",
      "state" : {
        "revision" : "4ab7d260301ddf1cab5376ea14e285a128f5ba39",
        "version" : "2.5.5"
      }
    },
    {
      "identity" : "gltfscenekit",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/magicien/GLTFSceneKit",
      "state" : {
        "branch" : "master",
        "revision" : "a9587ec5c6515ee7d036533fdd909815a9a718ef"
      }
    },
    {
      "identity" : "google-maps-ios-utils",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/googlemaps/google-maps-ios-utils.git",
      "state" : {
        "revision" : "55a9756a6fc8ea8f4a043f8c4109de516cd625ca",
        "version" : "3.10.3"
      }
    },
    {
      "identity" : "googleappmeasurement",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleAppMeasurement.git",
      "state" : {
        "revision" : "69f5b8425dd473d2f2475bba7f88f82cd9da0e36",
        "version" : "10.10.0"
      }
    },
    {
      "identity" : "googledatatransport",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleDataTransport.git",
      "state" : {
        "revision" : "5056b15c5acbb90cd214fe4d6138bdf5a740e5a8",
        "version" : "9.2.0"
      }
    },
    {
      "identity" : "googlesignin-ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleSignIn-iOS",
      "state" : {
        "revision" : "7932d33686c1dc4d7df7a919aae47361d1cdfda4",
        "version" : "7.0.0"
      }
    },
    {
      "identity" : "googleutilities",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleUtilities.git",
      "state" : {
        "revision" : "871d43135925cde39ef7421d8723ce47edfdcc39",
        "version" : "7.11.1"
      }
    },
    {
      "identity" : "grpc-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/grpc-binary.git",
      "state" : {
        "revision" : "2b7656d8ee892cd39db460b1a0d849de618405e6",
        "version" : "1.50.1"
      }
    },
    {
      "identity" : "gtm-session-fetcher",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/gtm-session-fetcher.git",
      "state" : {
        "revision" : "d415594121c9e8a4f9d79cecee0965cf35e74dbd",
        "version" : "3.1.1"
      }
    },
    {
      "identity" : "gtmappauth",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GTMAppAuth.git",
      "state" : {
        "revision" : "cee3c709307912d040bd1e06ca919875a92339c6",
        "version" : "2.0.0"
      }
    },
    {
      "identity" : "instabug-sp",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/Instabug/Instabug-SP.git",
      "state" : {
        "revision" : "498b2bb7dd418c8faf24f41ed20c48a3de977a7e",
        "version" : "10.13.1"
      }
    },
    {
      "identity" : "ios-branch-sdk-spm",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/BranchMetrics/ios-branch-sdk-spm.git",
      "state" : {
        "revision" : "6f3169ee741d18531f0bdbbcf900fd87b411bae7",
        "version" : "1.43.1"
      }
    },
    {
      "identity" : "keychainaccess",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/kishikawakatsumi/KeychainAccess",
      "state" : {
        "branch" : "master",
        "revision" : "6299daec1d74be12164fec090faf9ed14d0da9d6"
      }
    },
    {
      "identity" : "leveldb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/leveldb.git",
      "state" : {
        "revision" : "0706abcc6b0bd9cedfbb015ba840e4a780b5159b",
        "version" : "1.22.2"
      }
    },
    {
      "identity" : "libwebp-xcode",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SDWebImage/libwebp-Xcode.git",
      "state" : {
        "revision" : "0f3bdb28a1edc5e8e43876d3835d20c601ef331f",
        "version" : "1.2.3"
      }
    },
    {
      "identity" : "lottie-ios",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/airbnb/lottie-ios.git",
      "state" : {
        "revision" : "b4bd0604ded9574807f41b4004b57dd1226a30a4",
        "version" : "3.5.0"
      }
    },
    {
      "identity" : "marqueelabel",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/cbpowell/MarqueeLabel.git",
      "state" : {
        "revision" : "f2c72a5f8568579dade6350dc26a482076d3d346",
        "version" : "4.3.0"
      }
    },
    {
      "identity" : "mixpanel-swift",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/mixpanel/mixpanel-swift",
      "state" : {
        "revision" : "369e5796b2a166b4a8abb1c73c6014f5e02513bd",
        "version" : "3.5.1"
      }
    },
    {
      "identity" : "multicastdelegate",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/v-grigoriev/MulticastDelegate.git",
      "state" : {
        "revision" : "d650db9f294d2ebeab9e48e1825b6a16af860a62",
        "version" : "1.0.1"
      }
    },
    {
      "identity" : "nanopb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/nanopb.git",
      "state" : {
        "revision" : "819d0a2173aff699fb8c364b6fb906f7cdb1a692",
        "version" : "2.30909.0"
      }
    },
    {
      "identity" : "promisekit",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/mxcl/PromiseKit.git",
      "state" : {
        "revision" : "43772616c46a44a9977e41924ae01d0e55f2f9ca",
        "version" : "6.18.1"
      }
    },
    {
      "identity" : "promises",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/promises.git",
      "state" : {
        "revision" : "3e4e743631e86c8c70dbc6efdc7beaa6e90fd3bb",
        "version" : "2.1.1"
      }
    },
    {
      "identity" : "qrcode",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/WalletConnect/QRCode",
      "state" : {
        "revision" : "263f280d2c8144adfb0b6676109846cfc8dd552b",
        "version" : "14.3.1"
      }
    },
    {
      "identity" : "realm-cocoa",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/realm/realm-cocoa.git",
      "state" : {
        "revision" : "ae0ceea258fd8f58392881cc4b2228bff62f74f5",
        "version" : "10.28.7"
      }
    },
    {
      "identity" : "realm-core",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/realm/realm-core",
      "state" : {
        "revision" : "6f6a0f415bd33cf2ced4467e36a47f7c84f0a1d7",
        "version" : "12.5.1"
      }
    },
    {
      "identity" : "scnrecorder",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/gorastudio-ceo/SCNRecorder",
      "state" : {
        "revision" : "e4e1b7c579eff8feb4a93f2c9c31940a54cdcd07",
        "version" : "2.8.1"
      }
    },
    {
      "identity" : "sdwebimage",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SDWebImage/SDWebImage.git",
      "state" : {
        "revision" : "6c6b951845a520fa7e8356e28adb5339c0f008d3",
        "version" : "5.15.0"
      }
    },
    {
      "identity" : "sdwebimagepdfcoder",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SDWebImage/SDWebImagePDFCoder.git",
      "state" : {
        "revision" : "aec21a33971011dba672630c81cba5c44c7ec788",
        "version" : "0.8.0"
      }
    },
    {
      "identity" : "sdwebimagesvgcoder",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SDWebImage/SDWebImageSVGCoder.git",
      "state" : {
        "revision" : "afbe025cbdae37c20c144956fbabc07ddd5c0fc2",
        "version" : "1.6.1"
      }
    },
    {
      "identity" : "sdwebimagewebpcoder",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SDWebImage/SDWebImageWebPCoder.git",
      "state" : {
        "revision" : "8a0c5e1ae08ed763739262b9dcef64cfb241c14b",
        "version" : "0.9.0"
      }
    },
    {
      "identity" : "showtime",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/KaneCheshire/ShowTime.git",
      "state" : {
        "revision" : "f4e976931fc5f9b7a7d011f40c62a250c4ff53ab",
        "version" : "2.5.3"
      }
    },
    {
      "identity" : "snap-kit-spm",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/Snapchat/snap-kit-spm",
      "state" : {
        "revision" : "071281e1d9a1f3f6a875b105d866e2ab3d7c4234",
        "version" : "2.4.0"
      }
    },
    {
      "identity" : "snapkit",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SnapKit/SnapKit.git",
      "state" : {
        "revision" : "f222cbdf325885926566172f6f5f06af95473158",
        "version" : "5.6.0"
      }
    },
    {
      "identity" : "starscream",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/daltoniam/Starscream",
      "state" : {
        "revision" : "a063fda2b8145a231953c20e7a646be254365396",
        "version" : "3.1.2"
      }
    },
    {
      "identity" : "swift-algorithms",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/apple/swift-algorithms",
      "state" : {
        "revision" : "b14b7f4c528c942f121c8b860b9410b2bf57825e",
        "version" : "1.0.0"
      }
    },
    {
      "identity" : "swift-markdown-ui",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/gonzalezreal/swift-markdown-ui",
      "state" : {
        "revision" : "12b351a75201a8124c2f2e1f9fc6ef5cd812c0b9",
        "version" : "2.1.0"
      }
    },
    {
      "identity" : "swift-numerics",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/apple/swift-numerics",
      "state" : {
        "revision" : "0a5bc04095a675662cf24757cc0640aa2204253b",
        "version" : "1.0.2"
      }
    },
    {
      "identity" : "swift-protobuf",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/apple/swift-protobuf.git",
      "state" : {
        "revision" : "b8230909dedc640294d7324d37f4c91ad3dcf177",
        "version" : "1.20.1"
      }
    },
    {
      "identity" : "swift-qrcode-generator",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/dagronf/swift-qrcode-generator",
      "state" : {
        "revision" : "5ca09b6a2ad190f94aa3d6ddef45b187f8c0343b",
        "version" : "1.0.3"
      }
    },
    {
      "identity" : "swiftimagereadwrite",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/dagronf/SwiftImageReadWrite",
      "state" : {
        "revision" : "5596407d1cf61b953b8e658fa8636a471df3c509",
        "version" : "1.1.6"
      }
    },
    {
      "identity" : "swiftmarkdownkit",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/flickplay/SwiftMarkdownKit",
      "state" : {
        "branch" : "master",
        "revision" : "d76f95e3e9c8e0d80b8705c3902135d2631fd7c9"
      }
    },
    {
      "identity" : "swiftyjson",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/SwiftyJSON/SwiftyJSON.git",
      "state" : {
        "revision" : "b3dcd7dbd0d488e1a7077cb33b00f2083e382f07",
        "version" : "5.0.1"
      }
    },
    {
      "identity" : "visualeffectview",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/efremidze/VisualEffectView.git",
      "state" : {
        "revision" : "92f794bce4673e588f577b8477ec7a4f05f89222",
        "version" : "4.1.4"
      }
    },
    {
      "identity" : "walletconnectswiftv2",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/WalletConnect/WalletConnectSwiftV2",
      "state" : {
        "revision" : "7d4eca9cf4441bbfef649a055baea0a31fd96d83",
        "version" : "1.6.14"
      }
    },
    {
      "identity" : "ziparchive",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/ZipArchive/ZipArchive.git",
      "state" : {
        "revision" : "38e0ce0598e06b034271f296a8e15b149c91aa19",
        "version" : "2.4.3"
      }
    }
  ],
  "version" : 2
}


```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ncooke3** [comment]
Hi @tarrouye, this issue was addressed in GRPC `1.50.2`, but based on your `Package.resolved`, your project has `1.50.1`

```json
    {
      "identity" : "grpc-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/grpc-binary.git",
      "state" : {
        "revision" : "2b7656d8ee892cd39db460b1a0d849de618405e6",
        "version" : "1.50.1"
      }
    },
```

I suspect updating the dependency should resolve the issue. So in Xcode, select `File → Packages → Update to Latest Package Versions` and a newer GRPC version should be pulled in. 

This GRPC version with the fix should be compatible with Firebase 10.10.0:
https://github.com/firebase/firebase-ios-sdk/blob/5034479ca0c4f32f299677fb0ba1fe4d3e3e20b4/Package.swift#L169-L172

Let me know if this unblocks you or not!

**Comment by tarrouye** [comment]
Hi @ncooke3 , thanks for the response. I've already tried Update to Latest Package Versions, actually, and it still pulls 1.50.1 (confirmed again just now). Maybe it's a dependency from another Google package causing the issue and I've opened this issue in the wrong place 🤔 I'll try to take a closer look where the dependency is coming from
